exec/java-exec/src/test/java/org/apache/drill/exec/store/pcapng/TestPcapngHeaders.java

```diff
@@ -0,0 +1,154 @@
+    expectedSchema.add(MaterializedField.create("packet_data", Types.optional(TypeProtos.MinorType.VARCHAR)));
+    expectedSchema.add(MaterializedField.create("tcp_parsed_flags", Types.optional(TypeProtos.MinorType.VARCHAR)));
+    expectedSchema.add(MaterializedField.create("tcp_flags_ns", Types.optional(TypeProtos.MinorType.INT)));
+    expectedSchema.add(MaterializedField.create("src_port", Types.optional(TypeProtos.MinorType.INT)));
+    expectedSchema.add(MaterializedField.create("packet_length", Types.optional(TypeProtos.MinorType.INT)));
+    expectedSchema.add(MaterializedField.create("tcp_flags_urg", Types.optional(TypeProtos.MinorType.INT)));
+    expectedSchema.add(MaterializedField.create("tcp_ack", Types.optional(TypeProtos.MinorType.INT)));
+    expectedSchema.add(MaterializedField.create("dst_port", Types.optional(TypeProtos.MinorType.INT)));
+    expectedSchema.add(MaterializedField.create("dst_mac_address", Types.optional(TypeProtos.MinorType.VARCHAR)));
+
+    RowSet expected = new RowSetBuilder(client.allocator(), expectedSchema)
+        .build();
+    new RowSetComparison(expected)
+        .verifyAndClearAll(actual);
+  }
+
+  @Test
+  public void testValidHeadersForProjection() throws IOException {
+    String query = "select sRc_ip, dst_IP, dst_mAc_address, src_Port, tcp_session, `Timestamp`  from dfs.`store/pcapng/sniff.pcapng`";
+    RowSet actual = client.queryBuilder().sql(query).rowSet();
+
+    TupleSchema expectedSchema = new TupleSchema();
+
+    expectedSchema.add(MaterializedField.create("sRc_ip", Types.optional(TypeProtos.MinorType.VARCHAR)));
+    expectedSchema.add(MaterializedField.create("dst_IP", Types.optional(TypeProtos.MinorType.VARCHAR)));
+    expectedSchema.add(MaterializedField.create("dst_mAc_address", Types.optional(TypeProtos.MinorType.VARCHAR)));
+    expectedSchema.add(MaterializedField.create("src_Port", Types.optional(TypeProtos.MinorType.INT)));
+    expectedSchema.add(MaterializedField.create("tcp_session", Types.optional(TypeProtos.MinorType.BIGINT)));
+    expectedSchema.add(MaterializedField.create("Timestamp", Types.optional(TypeProtos.MinorType.TIMESTAMP)));
+
+    RowSet expected = new RowSetBuilder(client.allocator(), expectedSchema)
+        .build();
+    new RowSetComparison(expected)
+        .verifyAndClearAll(actual);
+  }
+
+  @Test
+  public void testValidHeadersForMissColls() throws IOException {
+    String query = "select `timestamp`, `name`, `color` from dfs.`store/pcapng/sniff.pcapng`";
+    RowSet actual = client.queryBuilder().sql(query).rowSet();
+
+    TupleSchema expectedSchema = new TupleSchema();
+
+    expectedSchema.add(MaterializedField.create("timestamp", Types.optional(TypeProtos.MinorType.TIMESTAMP)));
+    expectedSchema.add(MaterializedField.create("name", Types.optional(TypeProtos.MinorType.INT)));
+    expectedSchema.add(MaterializedField.create("color", Types.optional(TypeProtos.MinorType.INT)));
```

Comment:
I wonder, on this test case, was it the pcapng code that created the Nullable Int, or somewhere else in Drill?

Reply:
As I described above, Drill sets to all of the columns a MajorType as optional.